.kokoro/deps-finder.sh

```diff
@@ -0,0 +1,143 @@
+  latestSharedDeps=$2
+
+  if [[ -z ${latestSharedDeps} ]]; then
+    latestSharedDeps=$(getLatestSharedDepsVersion)
+    if [[ -z ${latestSharedDeps} ]]; then
+      echo "Unable to find correct version of shared-dependencies."
+      return 1
+    fi
+  fi
+
+  # Grab last part of dependency artifactId:version input.
+  # Remove any alphabetical characters at the end (e.g. -alpha, -beta, etc)
+  depVersion=${clientLibrary/*:/}
+  depVersion=${depVersion/-*/}
+
+  pomLocation="/pom.xml"
+
+  # The naming of these libraries does not follow the rules. As such, we've added in special cases for them.
+  # depArtifactId => the ID inside of the Github link for the dependency.
+  case ${clientLibrary} in
+  *"google-cloud-bigtable-bom"*)
+    depArtifactId="java-bigtable"
+    pomLocation="/google-cloud-bigtable-deps-bom/pom.xml"
+    ;;
+  *"google-cloud-core-bom"*)
+    # This is an exceptional case which does not require shared-dependencies! Return true automatically.
+    echo "Found exception google-cloud-core-bom!"
+    return 0
+    ;;
+  *"google-cloud-build-bom"*)
+    depArtifactId="java-cloudbuild"
+    ;;
+  *"google-cloud-monitoring-dashboard-bom"*)
+    depArtifactId="java-monitoring-dashboards"
+    ;;
+  *"google-cloud-nio"*)
+    depArtifactId="java-storage-nio"
+    ;;
+  *)
+    # General rules for obtaining the Github repo name are:
+    # Replace google-cloud with java, remove the version number, and remove -bom from the end (if there)
+    depArtifactId=${clientLibrary/*google-cloud-/java-}
+    depArtifactId=${depArtifactId/:*/}
+    depArtifactId=${depArtifactId/-bom*/}
+    ;;
+  esac
+  # Grab raw POM file from Github
+  rawFileStart="https://raw.githubusercontent.com/googleapis/"
+  gitPom=${rawFileStart}${depArtifactId}"/v"${depVersion}${pomLocation}
+
+  # If $3 is nonempty, we're running this script directly.
+  if [[ ! -z $3 ]]; then
+    echo "POM File Link Generated: ${gitPom}"
+  fi
+
+  content=$(wget ${gitPom} -q -O -)
+
+  if [[ -z $content ]]; then
+    echo "Failed to find POM for ${clientLibrary}!"
+    return 1
+  fi
+
+  # Grab all information from the client library's effective POM
+  currDepsVersion="$(echo "${content}" | sed '1,/google-cloud-shared-dependencies/d')"
+
+  # shared-dependencies is not in our dependencyManagement section!
+  if [[ -z $currDepsVersion ]]; then
+    echo "google-cloud-shared-dependencies is not listed for ${clientLibrary}!"
+    return 2
+  fi
+
+  currDepsVersion=${currDepsVersion/<\/version>*/}
+  currDepsVersion=${currDepsVersion/*<version>/}
+
+  if [[ ${currDepsVersion} != ${latestSharedDeps} ]]; then
+    echo "${clientLibrary} - Version Found: ${currDepsVersion}"
+    return 3
+  fi
+
+  echo "${clientLibrary} passed java-shared-dependencies check."
+  return 0
+}
+
+# If we're running the script directly from this file, we likely want to check the current commit, or skip.
+if [ "${BASH_SOURCE[0]}" -ef "$0" ]; then
+  echo $(checkLibrary "$1" "$2" 1)
+  exit $?
+fi
```

Comment:
A shell script should be either sourced or run, this pattern of checking to see if it was invoked directly is error-prone. If checkLibrary is used in multiple places, put it into the common functions script and source that script.

Reply:
In the new version, placed the function `checkLibrary` in the common functions script (shared-deps-helper.sh), removed the check to see which file the script was being run from, and the script is now run normally from deps-finder!